### PR TITLE
Bug 1323013: ignore 404 when cancel build requests

### PR DESCRIPTION
**Pull request by rail**
@Callek mind reviewing this?

### CODE REVIEW

**Comment by Callek** [comment]
What is the implications of this, as in, in what cases can a 404 happen with buildapi?

Early cancel (in which case buildbot will chug and TC will say canceled?) any others?

👍

**Comment by rail** [comment]
Another solution would be:
1) requeue the pulse message to the end of the queue
2) IIRC the message contains extra metadata when it's requeued, discard if it 404s again

This would add more complexity though...

**Comment by rail** [comment]
@catlee what do you think about this? The comments of `cancelBuildRequest()` mention 404, but without the explanation of possible root causes.

File: bbb/services.py [hunk 1]
```diff
@@ -595,7 +595,13 @@ def handleException(self, data, msg):
             # BuildRequest.
             else:
                 log.info("task %s: buildrequest %s: BuildRequest found for task, cancelling it.", taskid, brid)
-                self.selfserve.cancelBuildRequest(branch, brid)
+                try:
+                    self.selfserve.cancelBuildRequest(branch, brid)
+                except HTTPError as e:
+                    if e.response.status_code == 404:
+                        log.warn("Failed to cancel build request %s on %s. Ignoring the failure.", brid, branch)
```
Comment (catlee) [inline bbb/services.py]: please add the task id and buildrequest prefix here, to make log grepping easier

**Pull request merged**